src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -25,6 +25,17 @@ public partial class Ping : IDisposable
         private const int Disposed = 2;
         private int _status = Free;
 
+        public Ping()
+        {
+            if (GetType() == typeof(Ping)) // Not derived class. No use of finalizer here, suppress it immediately
```

Comment:
Please use braces for the if block. (First brace should start on the next line after if.)